lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -366,17 +367,9 @@ export class UserAgentApplication {
       // TODO: Should we throw noCallback error here?
     }
 
-    // TODO: Replace with new validation pattern
-    if (scopes) {
-      const isValidScope = this.validateInputScope(scopes);
-      if (isValidScope && !Utils.isEmpty(isValidScope)) {
-          if (this._tokenReceivedCallback) {
-              this._tokenReceivedCallback(ErrorDescription.inputScopesError, null, ErrorCodes.inputScopesError, Constants.idToken, this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie)));
-          return;
-        }
-      }
-      scopes = this.filterScopes(scopes);
-    }
+    // Validate and filter scopes (the validate function will throw if validation fails)
+    this.validateInputScope(scopes, false);
+    scopes = this.filterScopes(scopes);
 
```

Comment:
I know we have discussed this but I believe 'state' has to be a part of error response too as per: https://tools.ietf.org/html/rfc6749#section-10.12 
@manoj-rath Can you check the spec and comment?

Reply:
Prithvi and I looked at it. It looks like it matters only in case of successes, not errors because the state is useless if it failed. Also the exiting code was already returning the state in case of call back errors. So we decided to avoid state in case of clientValidation errors and return in case of callback errors.

Reply:
Okay! Sounds good.